lib/main.rb

```diff
@@ -0,0 +1,82 @@
+module Enumerable
+  def my_each
+    element = 0
+    while element < length
+      yield(self[element])
+      element += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    element = 0
+    while element < length
+      yield(self[element], element)
+      element += 1
+    end
+  end
+
+  def my_select
+    new_array = []
+    my_each do |a|
+      yield(a) ? new_array.push(a) : new_array
+    end
+  end
+
+  def my_all?
+    my_each do |a|
+      return false unless yield(a)
+    end
+    true
+  end
+
+  def my_any?
+    exist = false
+    my_each do |element|
+      exist = yield(element)
+      break if exist
+    end
+    exist
+  end
+
+  def my_none?
+    (my_any? { |i| yield(i) == true }) != true
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] my_none? when no block or argument is given returns true only if none of the collection members is true
```ruby
  array = [1,2,5,4,7]
  p array.my_none? == array.none? # false
```
- [ ] when a class is passed as an argument returns true if none of the collection is a member of such class
```ruby
  array = [1,2,5,4,7]
  p array.my_none?(Integer) == array.none?(Integer) # false
```
- [ ] when a Regex is passed as an argument returns true only if none of the collection matches the Regex
```ruby
  array = %w[dog door rod blade]
  p array.my_none?(/c/) == array.none?(/c/) # false
```
 - [ ] when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
```ruby
  array = %w[dog door rod blade]
  p array.my_none?(5) == array.none?(5) # false
```